bridges/snowbridge/pallets/system-v2/src/lib.rs

```diff
@@ -79,20 +74,19 @@ pub mod pallet {
 		/// Send messages to Ethereum
 		type OutboundQueue: SendMessage;
 
-		/// Origin check for XCM locations that can create agents
-		type SiblingOrigin: EnsureOrigin<Self::RuntimeOrigin, Success = Location>;
+		/// Origin check for XCM locations that transact with this pallet
+		type FrontendOrigin: EnsureOrigin<Self::RuntimeOrigin, Success = Location>;
 
 		/// Converts Location to AgentId
 		type AgentIdOf: ConvertLocation<AgentId>;
```

Comment:
Is there any reason why this type can't do reanchoring as well?

Also as I suggested in other comments, the word `Agent` is confusing. An agent is an ethereum contract. Within the context of this pallet, we are really just producing an `H256` to be able to set `message.origin`.

Whether the Gateway associates that with an Agent is not something this pallet should be concerned about.

```suggestion
		/// Converts Location to AgentId
		type LocationHashOf: ConvertLocation<H256>;
```